### PR TITLE
SWC-6264

### PR DESCRIPTION
**Pull request by yjyoo773**

### CODE REVIEW

**Comment by yjyoo773** [comment]
Screen Shot 2022-09-01 at 9 49 24 AM
Screen Shot 2022-09-01 at 9 49 31 AM
Screen Shot 2022-09-01 at 9 49 42 AM

Screen Shot 2022-09-01 at 9 50 31 AM

File: src/main/java/org/sagebionetworks/web/client/view/OAuthClientEditorViewImpl.java [hunk 1]
```diff
@@ -1,50 +1,37 @@
...
+    }
+
+    @UiField
+    SimplePanel componentContainer;
     private Header headerWidget;
 
     @Inject
-    public OAuthClientEditorViewImpl(Header headerWidget, SynapseContextPropsProvider propsProvider) {
-        container = new ReactComponentDiv();
+    public OAuthClientEditorViewImpl(OAuthClientEditorViewImplUiBinder binder, Header headerWidget) {
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/view/OAuthClientEditorViewImpl.java]: Why not inject the `SynapseContextPropsProvider` here?

File: src/main/java/org/sagebionetworks/web/client/view/OAuthClientEditorViewImpl.java [hunk 2]
```diff
@@ -1,50 +1,37 @@
...
-    @Override
-    public Widget asWidget() { return container.asWidget(); }
 
     @Override
-    public void render() {
-        EmptyProps props = EmptyProps.create();
-        ReactElement component = React.createElementWithSynapseContext(SRC.SynapseComponents.OAuthManagement, props, propsProvider.getJsInteropContextProps());
-        ReactDOM.render(component, container.getElement());
+    public void createReactComponentWidget(SynapseContextPropsProvider propsProvider) {
+        OAuthClientEditor component = new OAuthClientEditor(propsProvider);
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/view/OAuthClientEditorViewImpl.java]:
```suggestion
    public void createReactComponentWidget() {
        OAuthClientEditor component = new OAuthClientEditor(this.propsProvider);
```

Comment (nickgros) [reply src/main/java/org/sagebionetworks/web/client/view/OAuthClientEditorViewImpl.java]: See above

**Review by nickgros — COMMENTED** [review]
Small change--no need to pass the propsProvider to the view when the view can have it injected itself

File: src/main/java/org/sagebionetworks/web/client/presenter/OAuthClientEditorPresenter.java [hunk 3]
```diff
@@ -13,21 +13,23 @@ public class OAuthClientEditorPresenter extends AbstractActivity implements Pres
 
     private OAuthClientEditorView view;
     private OAuthClientEditorPlace place;
+    private SynapseContextPropsProvider propsProvider;
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/presenter/OAuthClientEditorPresenter.java]:
```suggestion
```

File: src/main/java/org/sagebionetworks/web/client/presenter/OAuthClientEditorPresenter.java [hunk 4]
```diff
@@ -13,21 +13,23 @@ public class OAuthClientEditorPresenter extends AbstractActivity implements Pres
...
 
     @Inject
-    public OAuthClientEditorPresenter(OAuthClientEditorView view, GlobalApplicationState globalApplicationState) {
+    public OAuthClientEditorPresenter(OAuthClientEditorView view, SynapseContextPropsProvider propsProvider) {
         this.view = view;
+        this.propsProvider = propsProvider;
     }
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/presenter/OAuthClientEditorPresenter.java]:
```suggestion
    public OAuthClientEditorPresenter(OAuthClientEditorView view) {
        this.view = view;
    }
```

**Review by nickgros — CHANGES_REQUESTED** [review]

**Review by nickgros — APPROVED** [review]

**Pull request merged**